ddl/ddl_api.go

```diff
@@ -2167,18 +2173,31 @@ func (d *ddl) RecoverTable(ctx sessionctx.Context, recoverInfo *RecoverInfo) (er
 		return infoschema.ErrTableExists.GenWithStackByArgs(tbInfo.Name)
 	}
 
-	tbInfo.State = model.StateNone
+	recoverInfo.TableInfo.State = model.StateNone
 	job := &model.Job{
 		SchemaID:   schemaID,
 		TableID:    tbInfo.ID,
 		SchemaName: schema.Name.L,
 		Type:       model.ActionRecoverTable,
 		BinlogInfo: &model.HistoryInfo{},
-		Args: []interface{}{tbInfo, recoverInfo.AutoIDs.RowID, recoverInfo.DropJobID,
-			recoverInfo.SnapshotTS, recoverTableCheckFlagNone, recoverInfo.AutoIDs.RandomID,
-			recoverInfo.OldSchemaName, recoverInfo.OldTableName},
+		Args:       []interface{}{recoverInfo, recoverCheckFlagNone},
```

Comment:
This refactor is great, but I am afraid it will break the compatibility, since the new DDL owner can't handle the old `ActionRecoverTable` ddl job now.

Reply:
cc @wjhuang2016

Reply:
I think it's acceptable. I don't think users would flashback database during upgrade.